yt/frontends/gadget/tests/test_outputs.py

```diff
@@ -124,3 +125,25 @@ def test_bigendian_field_access():
     ds = data_dir_load(BE_Gadget)
     data = ds.all_data()
     data["Halo", "Velocities"]
+
+
+mag_fields = OrderedDict(
```

Comment:
quick note: since Python 3.6, builtin dict objects are also ordered. At the time it was deemed an implementation detail, but it's been embraced as a feature since.
Some minor differences persist (see https://stackoverflow.com/questions/34305003/difference-between-dictionary-and-ordereddict), but it's doubtful we need them here

Reply:
So if you look at the top of the file:

https://github.com/yt-project/yt/blob/0c0ad20e93750e34b60ac9863098c9e561ac5ec6/yt/frontends/gadget/tests/test_outputs.py#L27-L35

We have another one of these in there, which I copied to make the new one below. For some reason there are two `("gas","temperature")` entries in here--`OrderedDict` simply replaces the first one with the second one silently without complaining. When I tried to change this to a `dict` the pre-commit bot complained. 

So the real question is if `sph_answer` is expecting us to try projections with and without the weight field for `("gas","temperature")`. Does anyone know (I can look myself, but if anyone knows please comment)? @matthewturk @brittonsmith @chummels

Reply:
It seems to me taking an unweighted projection of `("gas", "temperature")` would be useless and non-sensical, so I don't think `sph_answer` should do that.

Reply:
It probably is expecting that.  A non-weighted temperature projection is generally not useful, but it can also serve as a proxy for testing in situations where the temperature itself is a derived field (rather than intrinsic) and we want to test the machinery of unweighted temperature projections.  I don't know that we need to keep it in this instance, but I don't think we should *always* require that answer tests always map 1:1 to useful quantities.